cleverhans/experimental/certification/dual_formulation.py

```diff
@@ -50,6 +58,9 @@ def __init__(self, sess, dual_var, neural_net_param_object, test_input, true_cla
     self.input_minval = tf.convert_to_tensor(input_minval, dtype=tf.float32)
     self.input_maxval = tf.convert_to_tensor(input_maxval, dtype=tf.float32)
     self.epsilon = tf.convert_to_tensor(epsilon, dtype=tf.float32)
+    self.lzs_params = lzs_params
+    if not self.lzs_params:
+      self.lzs_params = DEFAULT_LZS_PARAMS
```

Comment:
It's better to make copy of DEFAULT_LZS_PARAMS here, i.e. `self.lzs_params = DEFAULT_LZS_PARAMS.copy()`

The reason is following. DEFAULT_LZS_PARAMS is not a real constant, it's just a convention that DEFAULT_LZS_PARAMS should be unchanged. So if value of `self.lzs_params` will be changed in this class then value of DEFAULT_LZS_PARAMS also will change.
It does not happen in your code, but it's possible that change of `self.lzs_params` will happen in some future modifications, which can lead to hard-to-find bugs.
To prevent such hard-to-find bugs, it's better to make a copy of DEFAULT_LZS_PARAMS